cpp/arcticdb/util/key_utils.hpp

```diff
@@ -41,12 +41,34 @@ inline void delete_keys_of_type_if(const std::shared_ptr<Store>& store, Predicat
     }
 }
 
+template<class Predicate>
+inline void delete_keys_of_type_if_sync(const std::shared_ptr<Store>& store, Predicate&& predicate, KeyType key_type, const std::string& prefix = std::string(), bool continue_on_error = false) {
```

Comment:
Just delete_keys_of_type_sync would be better

Reply:
as discussed, despite the name is getting a bit wierd, it is still consistent with the non-sync version